src/transformers/models/convnext/modeling_tf_convnext.py

```diff
@@ -0,0 +1,620 @@
+            kernel_initializer=get_initializer(config.initializer_range),
+            bias_initializer="zeros",
+        )
+        self.layernorm = tf.keras.layers.LayerNormalization(epsilon=1e-6, name="layernorm")
+
+    def call(self, pixel_values):
+        if isinstance(pixel_values, dict):
+            pixel_values = pixel_values["pixel_values"]
+
+        # When running on CPU, `tf.keras.layers.Conv2D` doesn't support `NCHW` format.
+        # So change the input format from `NCHW` to `NHWC`.
+        # shape = (batch_size, in_height, in_width, in_channels=num_channels)
+        pixel_values = tf.transpose(pixel_values, perm=(0, 2, 3, 1))
+
+        embeddings = self.patch_embeddings(pixel_values)
+        embeddings = self.layernorm(embeddings)
+        return embeddings
+
+
+class TFConvNextLayer(tf.keras.layers.Layer):
+    """This corresponds to the `Block` class in the original implementation.
+
+    There are two equivalent implementations: [DwConv, LayerNorm (channels_first), Conv, GELU,1x1 Conv]; all in (N, C,
+    H, W) (2) [DwConv, Permute to (N, H, W, C), LayerNorm (channels_last), Linear, GELU, Linear]; Permute back
+
+    The authors used (2) as they find it slightly faster in PyTorch. Since we already permuted the inputs to follow
+    NHWC ordering, we can just apply the operations straight-away without the permutation.
+
+    Args:
+        config ([`ConvNextConfig`]): Model configuration class.
+        dim (`int`): Number of input channels.
+        drop_path (`float`): Stochastic depth rate. Default: 0.0.
+    """
+
+    def __init__(self, config, dim, drop_path=0.0, **kwargs):
+        super().__init__(**kwargs)
+        self.dim = dim
+        self.config = config
+        self.dwconv = tf.keras.layers.Conv2D(
+            filters=dim,
+            kernel_size=7,
+            padding="same",
+            groups=dim,
+            kernel_initializer=get_initializer(config.initializer_range),
+            bias_initializer="zeros",
+            name="dwconv",
+        )  # depthwise conv
+        self.layernorm = tf.keras.layers.LayerNormalization(
+            epsilon=1e-6,
+            name="layernorm",
+        )
+        self.pwconv1 = tf.keras.layers.Dense(
+            units=4 * dim,
+            kernel_initializer=get_initializer(config.initializer_range),
+            bias_initializer="zeros",
+            name="pwconv1",
+        )  # pointwise/1x1 convs, implemented with linear layers
+        self.act = get_tf_activation(config.hidden_act)
+        self.pwconv2 = tf.keras.layers.Dense(
+            units=dim,
+            kernel_initializer=get_initializer(config.initializer_range),
+            bias_initializer="zeros",
+            name="pwconv2",
+        )
+        # Using `layers.Activation` instead of `tf.identity` to better control `training`
+        # behaviour.
+        self.drop_path = (
+            TFConvNextDropPath(
+                drop_path,
+                name="drop_path",
+            )
```

Comment:
```suggestion
            TFConvNextDropPath(drop_path, name="drop_path")
```

Reply:
> This line should not be present indeed.

Any suggested workarounds for this?

Reply:
Regarding https://github.com/huggingface/transformers/pull/15750#discussion_r812695898, I did run `make style` and the current formatting came from that.

Reply:
> > This line should not be present indeed.
> 
> Any suggested workarounds for this?

What if we don't add this line
```
        self.output_attentions = None
```
and it will keep the default value `False` from the parent class `PretrainedConfig`?

It's a little bit non-sense (because `output_attentions` has no meaning for `ConvNext`). But it will work (right?) and make the test pass for this moment.

Once this PR being merged,  Hugging Face could discuss the best way to deal with the pure Conv. models internally.

Reply:
> It's a little bit non-sense (because output_attentions has no meaning for ConvNext). But it will work (right?) and make the test pass for this moment.

I remember running the tests without that config specified. It was failing. Let me recheck again.